client/components/token-field/index.jsx

```diff
@@ -431,7 +455,15 @@ var TokenField = React.createClass( {
 	},
 
 	_deleteToken: function( token ) {
-		this.props.onChange( without( this.props.value, token ) );
+		const searchValue = 'object' === typeof token ? token.value : token;
```

Comment:
Given that I've seen this logic twice already, might make for a good utility function

```js
getTokenValue( token ) {
	if ( 'object' === typeof token ) {
		return token.value;
	}

	return token;
}
```